src/blob/index/bptree/node.rs

```diff
@@ -38,13 +38,15 @@ impl Node {
         Ok(meta_size + (keys_buf_size + offsets_buf_size) as u64)
     }
 
-    pub(super) fn binary_search_serialized(key: &[u8], buf: &[u8]) -> Result<usize, usize> {
+    pub(super) fn binary_search_serialized<K: Key>(key: &K, buf: &[u8]) -> Result<usize, usize> {
+        let key_size = K::LEN as usize;
         let mut l = 0i32;
-        let mut r: i32 = (buf.len() / key.len() - 1) as i32;
+        let mut r: i32 = (buf.len() / key_size - 1) as i32;
         while l <= r {
             let m = (l + r) / 2;
-            let offset = m as usize * key.len();
-            match key.cmp(&buf[offset..(offset + key.len())]) {
+            let offset = m as usize * key_size;
+            let k = buf[offset..(offset + key_size)].to_vec().into(); // TODO Allow keys to be based on slices
```

Comment:
:+1: It would be great to avoid copying into a vector

Reply:
Yet I think it should be implemented in another ticker, as it will require lifetime to be added to `Key` trait

Reply:
Have you measured the performance impact of this data copying? If it is large, we will need to implement support for slices in 2.0 milestone

Reply:
It's about 10 microseconds more on index search, according to benchmarks. While working with file indexes it's not noticeable at all.

Reply:
The relative perf change is more interesting in this case

Reply:
~10% more per call to search (from ~70us to ~80us)

Reply:
Ok, thanks! This regression is not too large, so we can fix it in a separate task